app/controllers/follows_controller.rb

```diff
@@ -25,6 +25,28 @@ def show
     end
   end
 
+  def bulk_show
+    skip_authorization
+
+    response = params[:ids].map(&:to_i).each_with_object({}) do |id, hsh|
+      hsh[id] = if current_user.id == id
+                  "self"
+                else
+                  following_them_check = FollowChecker.new(current_user, params[:followable_type], id).cached_follow_check
+                  following_you_check = FollowChecker.new(User.find_by(id: id), params[:followable_type], current_user.id).cached_follow_check
```

Comment:
I also think we should load users in bulk. Right now we do a "select * from users" for each user that's not the current user sent by the JS client, could we load all of them in one query?

Also, it seems to me that `FollowChecker` only requires a few columns from user to function, so we don't need to load the entire table, that could also be an optimization :)

Reply:
I think those are great optimizations to make but I would rather make them in additional PRs rather than changing a ton of behavior in this PR. The goal of this PR is to decrease the hits to the server from the view so we can tighten up our general site throttling which alone is a win. Beyond that I am inclined to use separate PRs to get further optimizations.